images/kube-vip/README.md

```diff
@@ -0,0 +1,124 @@
+<!--monopod:start-->
+# kube-vip
+| | |
+| - | - |
+| **OCI Reference** | `cgr.dev/chainguard/kube-vip` |
+
+
+* [View Image in Chainguard Academy](https://edu.chainguard.dev/chainguard/chainguard-images/reference/kube-vip/overview/)
+* [View Image Catalog](https://console.enforce.dev/images/catalog) for a full list of available tags.
+* [Contact Chainguard](https://www.chainguard.dev/chainguard-images) for enterprise support, SLAs, and access to older tags.*
+
+---
+<!--monopod:end-->
+
+<!--overview:start-->
+Kubernetes Control Plane Virtual IP and Load-Balancer
+<!--overview:end-->
+
+<!--getting:start-->
+## Download this Image
+The image is available on `cgr.dev`:
+
+```
+docker pull cgr.dev/chainguard/kube-vip:latest
+```
+<!--getting:end-->
+
+<!--body:start-->
+
+## Usage
+
+You can test the kube-vip in several ways. Here is an example of how to deploy the kube-vip controller into KinD (Kubernetes in Docker) cluster:
```

Comment:
Is there a step we're missing to run this in privileged mode for local testing? When I follow these steps i'm just getting this in the log:

```bash
time="2024-04-30T11:21:38Z" level=warning msg="could not add ip: operation not permitted"
time="2024-04-30T11:21:38Z" level=warning msg="failed to get raw socket: operation not permitted"
```

Looks like in your nginx example you got it running however?

Reply:
yes during the nginx example I was using the docker commands, but for the automated tests I use helm chart, maybe there is something missing in the Helm chart for sure, because either documentation or the helm chart seem up to date to be honest 🙈

Reply:
https://github.com/kube-vip/helm-charts/blob/5f24093899db53f7c103bd95b0e41a112bbfb72b/charts/kube-vip/values.yaml#L65-L69 they seem set capabilities though 🤔

Reply:
What are the manual test steps we need to run to validate this in that case? it looks like its not coming up cleanly in the logs when I run these steps, so i don't have a high confidence that functionality is working, even when manually testing

Reply:
thread no longer applicable - we changed the test steps since this draft